lib/spec/sww.sw-spec.js

```diff
@@ -0,0 +1,122 @@
+importScripts('/base/lib/spec/nodeMock.js');
+importScripts('/base/lib/sww.js');
+importScripts('/base/lib/router.js');
+
+describe('ServiceWorkerWare', function () {
+  'use strict';
+
+  var worker;
+  var request;
+
+  var fallbackResult = 'fallbackMW';
+  var fallbackMW = function(req, res) {
+    return fallbackResult;
+  }
+
+  beforeEach(function() {
+    worker = new ServiceWorkerWare();
+    worker.init();
+    request = {
+      method: 'GET',
+      url: 'https://www.example.com',
+      clone: function() {}
+    };
+  });
+
+  describe('fallback mw', function(done) {
+
+    it('should fetch if no fallback middleware', function (done) {
+      var result = 'ok';
+      var stub = sinon.stub(self, 'fetch').returns(Promise.resolve(result));
+
+      worker.onFetch({
+        request: request,
+        respondWith: function(chain) {
```

Comment:
[What I did to solve this](https://github.com/lodr/serviceworkerware/blob/new-middleware-api/lib/spec/sww.sw-spec.js#L40).